services/venice-router/src/main/java/com/linkedin/venice/router/stats/RouterHttpRequestStats.java

```diff
@@ -168,8 +159,7 @@ public RouterHttpRequestStats(
     Rate tardyRequestRate = new OccurrenceRate();
 
     incomingRequestSensor = registerSensor("request", new Count(), requestRate);
-    incomingRequestSensorOtel = (LongCounter) metricsRepository.getOpenTelemetryMetricsRepository()
-        .getInstrument(INCOMING_CALL_COUNT.getMetricEntity());
+    INCOMING_CALL_COUNT.getMetricEntity().createMetric(otelRepository);
```

Comment:
This looks problematic to me... here we are priming the singleton state of the enum.

In theory it works, but typically we consider static mutable state to be an anti-pattern. It can make testing more difficult, e.g. by causing two router instances in the same JVM to share state among them, or by having one test pollute the state of a subsequent test.

I think we should either make `MetricEntity` some kind of immutable definition, which does not carry any operational state, and from which the operational instance (likely some other class) is created. Or else if we wish `MetricEntity` to be the operational class, then we should make the enum carry a constructor of `MetricEntity` rather than the `MetricEntity` itself.

Reply:
As discussed, removed the operational state from `MetricEntity` and created a new class called `MetricEntityState` which contains `MetricEntity` object and  otel and a list of tehuti metrics. This class is instantiated for every metric. This still has some boiler plate like the Dimension value and the tehuti metric name while recording the metrics.